docs/book/stacks-and-components/auth-management/hyperai-service-connector.md

```diff
@@ -0,0 +1,50 @@
+---
+description: Configuring HyperAI Connectors to connect ZenML to HyperAI instances.
+---
+
+# HyperAI Service Connector
+
+The ZenML HyperAI Service Connector allows authenticating with a HyperAI instance for deployment of pipeline runs. This connector provides pre-authenticated Paramiko SSH clients to Stack Components that are linked to it.
+
+```
+$ zenml service-connector list-types --type hyperai
+┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┯━━━━━━━━━━━━┯━━━━━━━━━━━━━━━━━━━━┯━━━━━━━━━━━━━━┯━━━━━━━┯━━━━━━━━┓
+┃           NAME            │ TYPE       │ RESOURCE TYPES     │ AUTH METHODS │ LOCAL │ REMOTE ┃
+┠───────────────────────────┼────────────┼────────────────────┼──────────────┼───────┼────────┨
+┃ HyperAI Service Connector │ 🤖 hyperai │ 🤖 hyperai-generic │ rsa-key      │ ✅    │ ✅     ┃
+┃                           │            │                    │ dsa-key      │       │        ┃
+┃                           │            │                    │ ecdsa-key    │       │        ┃
+┃                           │            │                    │ ed25519-key  │       │        ┃
+┗━━━━━━━━━━━━━━━━━━━━━━━━━━━┷━━━━━━━━━━━━┷━━━━━━━━━━━━━━━━━━━━┷━━━━━━━━━━━━━━┷━━━━━━━┷━━━━━━━━┛
+```
+
+## Prerequisites
+The HyperAI Service Connector is part of the HyperAI integration. It is necessary to install the integration in order to use this Service Connector:
+
+* `zenml integration install hyperai` installs the HyperAI integration
+
+## Resource Types
+The HyperAI Service Connector supports HyperAI instances.
+
+## Authentication Methods
+ZenML creates an SSH connection to the HyperAI instance in the background when using this Service Connector. It then provides these connections to stack components requiring them, such as the HyperAI Orchestrator. Multiple authentication methods are supported:
+
+1. RSA key based authentication.
+2. DSA (DSS) key based authentication.
+3. ECDSA key based authentication.
+4. ED25519 key based authentication.
+
```

Comment:
It might be worth adding a "warning" hint here making users aware that the SSH private keys configured in the connector will be distributed to all clients that use them to run pipelines with the HyperAI orchestrator.

This is an obvious security weakness, as the SSH keys are long-lived credentials that give unrestricted access to the HyperAI instances.